packages/base/src/hooks/useI18nBundle.ts

```diff
@@ -1,20 +1,40 @@
 import { fetchI18nBundle, getI18nBundle } from '@ui5/webcomponents-base/dist/i18nBundle';
 import { useEffect, useState } from 'react';
 
+type TextWithDefault = { key: string; defaultText: string };
+type TextWithPlaceholders = [TextWithDefault, ...string[]];
+
 interface I18nBundle {
-  getText: (textObj: { key: string; defaultText: string }, ...args: any) => string;
+  getText: (textObj: TextWithDefault, ...args: any) => string;
 }
 
-export const useI18nBundle = (bundleName): I18nBundle => {
-  const [bundle, setBundle] = useState(getI18nBundle(bundleName));
+const resolveTranslations = (bundle, texts) => {
+  return texts.map((text) => {
+    if (Array.isArray(text)) {
+      const [key, ...replacements] = text;
+      return bundle.getText(key, replacements);
+    }
+    return bundle.getText(text);
+  });
+};
+
+export const useI18nText = (bundleName: string, ...texts: (TextWithDefault | TextWithPlaceholders)[]): string[] => {
+  const i18nBundle: I18nBundle = getI18nBundle(bundleName);
+  const [translations, setTranslations] = useState(resolveTranslations(i18nBundle, texts));
 
   useEffect(() => {
     const fetchAndLoadBundle = async () => {
       await fetchI18nBundle(bundleName);
-      setBundle(getI18nBundle(bundleName));
+      setTranslations((prev) => {
+        const next = resolveTranslations(i18nBundle, texts);
+        if (prev.length === next.length && prev.every((translation, index) => next[index] === translation)) {
+          return prev;
+        }
+        return next;
+      });
     };
     fetchAndLoadBundle();
-  }, []);
+  }, [fetchI18nBundle]);
```

Comment:
Please add dependencies.
```suggestion
  }, [fetchI18nBundle, bundleName, texts]);
```